populator-machinery/controller.go

```diff
@@ -549,8 +624,42 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	}
 
 	// *** Here is the first place we start to create/modify objects ***
+	// If the PVC is unbound and PVC' doesn't exist yet, create PVC'
+	if "" == pvc.Spec.VolumeName && pvcPrime == nil {
+		pvcPrime = &corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pvcPrimeName,
+				Namespace: c.populatorNamespace,
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      pvc.Spec.AccessModes,
+				Resources:        pvc.Spec.Resources,
+				StorageClassName: &storageClassPrimeName,
+				VolumeMode:       pvc.Spec.VolumeMode,
+			},
+		}
+		if waitForFirstConsumer {
+			pvcPrime.Annotations = map[string]string{
+				annSelectedNode: nodeName,
+			}
+		}
+		_, err = c.kubeClient.CoreV1().PersistentVolumeClaims(c.populatorNamespace).Create(ctx, pvcPrime, metav1.CreateOptions{})
```

Comment:
you can capture the newly created pvCPrime in L646 and use it to move forward to L656 if there is no error